### ISSUE TITLE
Allow way to modify Compose template

### ISSUE THREAD

**Issue by ChaosExAnima** [10up/wp-local-docker-v2#18]
It would be useful for less than standard setups to provide a way to modify the compose template via a file in the config directory, which is deep-merged on the create command.

My specific use case is that I need to specify the default network's subnet range, or it cannot be created as expected. I can see there could be other use cases, however.

Would this be something worth a PR?

**Comment by rocketcrazy07** [comment]
I agree with this and would go one step further.  I use nginx for other development and thus anytime I want to use this I have to shut down my local nginx to allow the gateway to start.  I would add an advanced option or something to allow the use of a local ngnix or at the very least be able to change the port that the gateway binds to on the host machine.  This would allow me to setup my own local nginx to act as the proxy to either the 10up gateway proxy or the enviroments nginx.

**Comment by eugene-manuilov** [comment]
This will be possible once the next version `2.8.0` is released. It will include a command to create a config file for a local environment. That config file can be committed into repository and used to create new environments on other computers. The config file will allow you to define a `dockerCompose` function that receives the original docker-compose.yml file (as a json object) and should return updated version.

Here is an example which replaces memcached with redis: 
```js
exports.dockerCompose = function( baseConfig ) {
    const config = { ...baseConfig };

    // replace memcached with redis service
    config.services.phpfpm.depends_on.push( 'redis' );
    config.services.phpfpm.depends_on = config.services.phpfpm.depends_on.filter( ( service ) => service !== 'memcached' );
    config.services.redis = { image: 'redis:latest' };
    delete config.services.memcached;

    return config;
};
```

To make use of it, you will need to create a new environment by running `10updocker clone {giturl}` command. For example:

```sh-session
10updocker clone git@github.com:eugene-manuilov/woocommerce.git
```

**Comment by eugene-manuilov** [comment]
It has been implemented and merged via #123. We can consider this ticket as done.
